substrate-node/pallets/pallet-dao/src/proposal.rs

```diff
@@ -0,0 +1,25 @@
+use codec::{Decode, Encode};
+use frame_support::traits::Vec;
+
+#[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Debug, Encode, Decode, Default)]
+pub struct DaoProposal<ProposalIndex> {
+    pub index: ProposalIndex,
+    pub description: Vec<u8>,
+    pub link: Vec<u8>,
+}
+
+#[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Debug, Encode, Decode, Default)]
+pub struct Votes<ProposalIndex, BlockNumber, AccountId> {
+    pub index: ProposalIndex,
+    pub threshold: u32,
+    pub ayes: Vec<VoteWeight>,
+    pub nays: Vec<VoteWeight>,
+    pub end: BlockNumber,
+    pub vetos: Vec<AccountId>
+}
+
+#[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Debug, Encode, Decode, Default)]
+pub struct VoteWeight {
+    pub farm_id: u32,
+    pub weight: u64,
```

Comment:
We shouldn't include the weight, instead only track the farms voting position. This way we get up to date weights when close is called (we will collect weights there).

Reply:
Not sure if collecting weights at close is the best option for performance. This can be a heavy storage operation to retrieve all the weights for all the farmers that have voted during a proposal.

Reply:
Not sure if collecting weights at close is the best option for performance. This can be a heavy storage operation to retrieve all the weights for all the farmers that have voted during a proposal.

Reply:
Let's open an issue for a feature improvement for this. If we want to collect weights at close we need to properly set the weight for this extrinsic (needs to be dynamic over the number of farmers voted)